MailCore/Cache/BackgroundRealm.swift

```diff
@@ -38,9 +38,12 @@ public final class BackgroundRealm {
 
     public func execute<T>(_ block: @escaping (Realm) -> T, completion: @escaping (T) -> Void) {
         BackgroundExecutor.executeWithBackgroundTask { [weak self] taskCompleted in
-            self?.queue.async {
-                guard let realm = self?.getRealm() else { return }
-                realm.refresh()
+            guard let self else {
+                return
+            }
+
+            queue.async {
```

Comment:
Do we want to keep a reference to self to execute the async block even if we try to deinit the whole object ?

Reply:
Correct I applied the weak / strong self pattern by reflex, as it is usualy cleaner.

Yeah I guess we can keep the double unwrap, se we can exit within the serial queue.